### PR TITLE
[CU-8686x1vdp] Adds support for auto generated license addendum

### PR DESCRIPTION
**Pull request by manuelurenah**

### CODE REVIEW

File: src/components/Model/ModelVersions/ModelVersionDetails.tsx [hunk 1]
```diff
@@ -450,6 +450,15 @@ export function ModelVersionDetails({
       : `Removal reason: ${version.meta?.customMessage}.` ?? '';
   const license = baseModelLicenses[version.baseModel];
   const onSite = !!version.trainingStatus;
+  const showAddendumLicense =
+    ['SD 1.5', 'SDXL 1.0'].includes(version.baseModel) &&
```
Comment (lrojas94) [inline src/components/Model/ModelVersions/ModelVersionDetails.tsx:454]: Might be good to keep this list of values in a constant for easy access

File: src/components/PermissionIndicator/PermissionIndicator.tsx [hunk 2]
```diff
@@ -13,20 +13,22 @@ import {
 } from '@tabler/icons-react';
 import React from 'react';
 
+const canSellImagesPermissions = ['Image', 'Rent', 'RentCivit', 'Sell'] as const;
+const canRentCivitPermissions = ['RentCivit', 'Rent', 'Sell'] as const;
+const canRentPermissions = ['Rent', 'Sell'] as const;
+const canSellPermissions = ['Sell'] as const;
```
Comment (lrojas94) [inline src/components/PermissionIndicator/PermissionIndicator.tsx:19]: Curious why you're not using the Prisma enum values here

File: src/server/services/model-version.service.ts [hunk 3]
```diff
@@ -627,3 +629,89 @@ export const modelVersionGeneratedImagesOnTimeframe = async ({
...
+  }
+
+  if (!options.allowCommercialUse.includes(CommercialUse.Image))
+    license += commercialUsePermissionContent[CommercialUse.Image];
+  if (!options.allowCommercialUse.includes(CommercialUse.Rent))
+    license += commercialUsePermissionContent[CommercialUse.Rent];
+  if (!options.allowCommercialUse.includes(CommercialUse.RentCivit))
+    license += commercialUsePermissionContent[CommercialUse.RentCivit];
+  if (!options.allowCommercialUse.includes(CommercialUse.Sell))
+    license += commercialUsePermissionContent[CommercialUse.Sell];
```
Comment (lrojas94) [inline src/server/services/model-version.service.ts:714]: Just an opinion here but I think this works better as an array in which you push strings to and then join 🤔. Don't mind it too much, I just think it's better than adding up the string until the end.

File: src/server/services/model.service.ts [hunk 4]
```diff
@@ -378,23 +378,12 @@ export const getModelsRaw = async ({
...
-      case CommercialUse.Sell:
-        commercialUseOr.push(CommercialUse.Sell);
-    }
-
-    AND.push(Prisma.sql`m."allowCommercialUse" IN (${Prisma.join(commercialUseOr, ',')})`);
+  if (allowCommercialUse && allowCommercialUse.length > 0) {
+    AND.push(
+      Prisma.sql`m."allowCommercialUse" && ARRAY[${Prisma.join(
+        allowCommercialUse
+      )}]::"CommercialUse"[]`
```
Comment (lrojas94) [inline src/server/services/model.service.ts:385]: Doesn't `Prisma.join` require the `,` as a 2nd argument?

**Review by lrojas94 — APPROVED** [review]

File: src/components/PermissionIndicator/PermissionIndicator.tsx (hunk 2 above)
Comment (manuelurenah) [reply src/components/PermissionIndicator/PermissionIndicator.tsx:19]: Just to save a few key strokes 😛

**Pull request merged**